src/js/popup/popup.js

```diff
@@ -475,85 +476,14 @@
-              }
-              
-              const maskListItemAddress = document.createElement("div");
-              maskListItemAddress.classList.add("fx-relay-mask-item-address");
-              maskListItemAddress.textContent = mask.full_address;
-              maskListItemAddressWrapper.appendChild(maskListItemAddress);
-
-              // Add Mask Address Bar Contents 
-              maskListItemAddressBar.appendChild(maskListItemAddressWrapper);
-
-              const maskListItemAddressActions = document.createElement("div");
-              maskListItemAddressActions.classList.add("fx-relay-mask-item-address-actions");
-
-              const maskListItemCopyButton = document.createElement("button");
-              maskListItemCopyButton.classList.add("fx-relay-mask-item-address-copy");
-              maskListItemCopyButton.setAttribute("data-mask-address", mask.full_address);
-
-              const maskListItemCopyButtonSuccessMessage = document.createElement("span");
-              maskListItemCopyButtonSuccessMessage.textContent = browser.i18n.getMessage("popupCopyMaskButtonCopied");
-              maskListItemCopyButtonSuccessMessage.classList.add("fx-relay-mask-item-address-copy-success");
-              maskListItemAddressActions.appendChild(maskListItemCopyButtonSuccessMessage);
-              
-              maskListItemCopyButton.addEventListener("click", (e)=> {
-                e.preventDefault();
-                navigator.clipboard.writeText(e.target.dataset.maskAddress);
-                maskListItemCopyButtonSuccessMessage.classList.add("is-shown");
-                setTimeout(() => {
-                  maskListItemCopyButtonSuccessMessage.classList.remove("is-shown")
-                }, 1000);
-              }, false);
-              maskListItemAddressActions.appendChild(maskListItemCopyButton);
-
-              const maskListItemToggleButton = document.createElement("button");
-              maskListItemToggleButton.classList.add("fx-relay-mask-item-address-toggle");
-              maskListItemToggleButton.addEventListener("click", ()=> {
-                // TODO: Add Toggle Function
-              }, false);
-              maskListItemToggleButton.setAttribute("data-mask-id", mask.id);
-              maskListItemToggleButton.setAttribute("data-mask-type", mask.mask_type);
-              maskListItemToggleButton.setAttribute("data-mask-address", mask.full_address);
-
-              // TODO: Add toggle button back
-              // maskListItemAddressActions.appendChild(maskListItemToggleButton);
-
-              maskListItemAddressBar.appendChild(maskListItemAddressActions);
-              maskListItem.appendChild(maskListItemAddressBar);
-              maskList.appendChild(maskListItem);
+            
+            // Generate and append each mask item to the mask list
+            masks.forEach( mask => {
+                const maskListItem = popup.panel.masks.utilities.buildMaskListItem(mask);
```

Comment:
Context: This function was getting really big. Broke out the `buildMaskListItem` iteration to make it easier to read.